Decrypt Android Push Notifications

fpohtmeh-github opened the pull request.

fpohtmeh-github committed:
Added platform-notifications library

fpohtmeh-github committed:
Added placeholder for pushy.me notification service

SergeySeroshtan committed:
Merge branch 'develop' into feature/pushy-me-integration

# Conflicts:
#	customers/Area52/customer.cmake
#	customers/Braes/customer.cmake
#	customers/Virgil/customer.cmake

SergeySeroshtan committed:
Improve Push Notifications API

SergeySeroshtan committed:
Fix redundant push notifications - reduce XMPP room subscriptions

SergeySeroshtan committed:
Add decryption for Android push notifications

SergeySeroshtan committed:
Add decryption for Android push notifications - fix Java classes

SergeySeroshtan committed:
Switch Qt sources version to 5.15 within Android resources

SergeySeroshtan committed:
Add decryption for Android push notifications - fix Java / Qt Service

SergeySeroshtan committed:
Add decryption for Android push notifications - fix Java / Qt Service

SergeySeroshtan committed:
Add decryption for Android push notifications - fix message id

SergeySeroshtan committed:
Fix redundant push notifications - reduce XMPP room subscriptions 2

SergeySeroshtan committed:
Add decryption for Android push notifications - show placeholder message if failed to decrypt it

SergeySeroshtan committed:
Add decryption for Android push notifications - small code chnages

SergeySeroshtan committed:
Add decryption for Android push notifications - add filtering for group self messages

andrii-iakovenko commented on platforms/android/AndroidManifest.xml.in:
```diff
@@ -93,7 +93,12 @@
 
-        <!-- For adding service(s) please check: https://wiki.qt.io/AndroidServices -->
+
+        <service android:name="org.virgil.notification.FirebaseMessageReceiver">
```
I would use `org.virgil.messenger` or `@VS_ANDROID_PACKAGE_NAME@` package as a base. So that service name be the next `org.virgil.messenger.notification.FirebaseMessageReceiver`

SergeySeroshtan committed:
Do not export Android service PushHandlerService to avoid name collisions

SergeySeroshtan committed:
Show push messages for groups as "Encrypted (group)"

SergeySeroshtan committed:
Send 'received' marker only for peer-to-peer chats

SergeySeroshtan committed:
Run Android service that handles push notifications in a separate process

Motivation:
    When main application is running and when push notification comes that runs service which tries to load Qt libraries that were already loaded and this crashes the main application.
